src/go/rpk/pkg/cli/cluster/storage/unmount.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newUnmountCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "unmount [TOPIC]",
+		Short: "Unmount a topic from the Redpanda Cluster",
+		Long: `Unmount a topic from the Redpanda cluster and secure it in Tiered
+Storage.
+
+This command performs an operation that:
+1. Rejects all writes to the topic
+2. Flushes data to Tiered Storage
+3. Removes the topic from the cluster
+
+The unmount process ensures data safety and cluster integrity. Every
+acknowledged message before the initiation of unmount is guaranteed to be
+persisted in Tiered Storage, subject to Redpanda's existing data safety
+guarantees. 
+
+Key Points:
+- During unmounting, any attempted writes or reads will receive an
+  UNKNOWN_TOPIC_OR_PARTITION error.
+- The unmount operation works independently of other topic configurations like
+  remote.delete=false.
+- After unmounting, the topic can be remounted to this cluster or a different
+  cluster if the log segments are moved to that cluster's Tiered Storage.
+`,
+		Example: `Unmount topic 'my-topic' from the cluster in the 'my-namespace'
+  rpk cluster storage unmount my-namespace/my-topic
```

Comment:
```suggestion
		Example: `
Unmount topic 'my-topic' from the cluster in the 'my-namespace'
  rpk cluster storage unmount my-namespace/my-topic
```

This is to avoid
```
Usage:
  rpk cluster storage unmount [flags]

Example:
Unmount topic 'my-topic' from the cluster in the 'my-namespace
```